airflow/providers/amazon/provider.yaml

```diff
@@ -81,23 +81,14 @@ dependencies:
   - apache-airflow-providers-common-sql>=1.3.1
   - apache-airflow-providers-http
   # We should update minimum version of boto3 and here regularly to avoid `pip` backtracking with the number
-  # of candidates to consider. We should also make sure that all the below related packages have also the
-  # same minimum version specified. Boto3 1.28.0 has been released on July 6 2023. We should also make sure we
-  # set it to the version that `aiobotocore` supports (see `aiobotocore` optional dependency at the end
-  # of this file). Currently we set aiobotocore as minimum 2.5.3 - as this is was the first version
-  # that supported boto3 1.28.
-  # NOTE!!! BEFORE botocore 1.33 BOTOCORE VERSIONS ARE SHIFTED BY 3 MINOR VERSIONS
-  # NOTE!!! Make sure to update _MIN_BOTO3_VERSION in setup.py when you update it here
-  - boto3>=1.28.0
-  # NOTE!!! BEFORE botocore 1.33 version is always shifted by 3 MINOR VERSIONS from boto3, see:
-  # - https://github.com/boto/boto3/issues/2702
-  # - https://github.com/boto/boto3/blob/a3e710df58676c2567b8023dbe09025ad5e0e50f/.changes/1.33.0.json#L13-L15
-  - botocore>=1.31.0
+  # of candidates to consider. Make sure to configure boto3 version here as well as in all the tools below
+  # in the `devel-dependencies` section to be the same minimum version.
+  - boto3>=1.33.0
+  - botocore>=1.33.0
   - inflection>=0.5.1
-  # Allow a wider range of watchtower versions for flexibility among users
-  - watchtower>=2.0.1,<4
+  - watchtower>=3.0.0,<4
```

Comment:
Noting that we explicitly wanted longer range see https://github.com/apache/airflow/pull/35713
cc @o-nikolas 

reinstating version >=3 will bring back user pain that we wanted to avoid as explained in the PR.

Reply:
Yeah. I've been hesitating about that one too (for same reasons). But:

1) it's already > 1 year old
2) watchtower 2 is > 2 years old
3) users will still be able to use the providers (incluiding the one being voted now) with watchtower 2

We can **just** drop the limitation bump for watchtower from that one if that's a **real** problem - but I would at least want to know if there is a good reason for people to stick to watchtower 2. Because we could also treat it as a way to **make** people upgrade to watchtower 3 eventually if they are doing it mostly due to laziness and have no good reason for it.

Reply:
> but I would at least want to know if there is a good reason for people to stick to watchtower 2

Lets hear from @o-nikolas @shubham22

Reply:
Yeah, this is a particular quirk with how MWAA works with Watchtower. In this very particular case limiting to v3 causes a lot of user/customer pain. If we could avoid updating this one for now that would be greatly appreciated. I can bring back this issue to the MWAA team and they can work on a fix (hopefully soon)

Reply:
cc: @vincbeck @o-nikolas @ferruzzi  -> would love to hear from you on that one :D

Reply:
Right. No problem I think for now. I can remove this bump and get it for the next try....

(Quoting a famous role .... I'LL BE BACK)

Reply:
BTW. Woudl be great to have a reference to an issue (?) opened somewhere about it (maybe?) This way we can refer to it in the comment and not to repeat it next time, until it is solved.

Reply:
> BTW. Woudl be great to have a reference to an issue (?) opened somewhere about it (maybe?)

The context so far is this Slack thread where the issue was reported: https://apache-airflow.slack.com/archives/CCRR5EBA7/p1699981307610129

And my PR "fixing" the issue: https://github.com/apache/airflow/pull/35713

Reply:
> Is there any reason we want to limit the watchtower version to v3.x? Is it required for updating any other dependency?

While testing the new packaging for Airflow in https://github.com/apache/airflow/pull/36537 I just noticed that there was a long backtracking from `pip` and I decided to just upgrade a few obvious candidates. I did it one-by -one and I saw each of them was causing shorter back-tracking  - I saw watchtower as one of the culprits, so I bumped it while testing. 

But eventually, I think watchtower is not the major contributor - it adds a little, But I think there were other reasons (some of teething problems in the new packaging that I fixed this afternoon) . I think the other bumps (especially botocore) could be the main cause for "longer" backtracking so I think it's not an issue to leave watchtower limited to 2.0.1. 

I will see when I merge all the oustanding PRs and rebase #36537 - this can be easily seen by the time "upgrade to newer dependencies" take. Particularly this step: https://github.com/apache/airflow/actions/runs/7449867636/job/20268244835?pr=36537 - 

It takes 2m20s for my last attempt. If we can keep it to few minutes rather than 20/30 then it is completely no issue.

Reply:
The problem with those backtracking issues that you NEVER know for sure unless you try all combinations. Sometimes just bumping either this or that makes a huge difference and you have to choose which one is "more reasonable". In this case I think the problem is relation between `botocore` and `watchtower` so as long as I will bump one of them and problem is gone, I am happy.

Reply:
Thanks for going in the details. That helps. I will also keep track of the installation time on your PR. 
From MWAA side, it is preferred to delay the watchtower dependency change for at least 5-6 months, so users on v2.7.2 (and older) have had at least 6 months to migrate over to the newer versions. In any case, we will start supporting it with our next release of v2.8.

Reply:
Maybe worth to mention that the mixed governance model https://github.com/apache/airflow/blob/main/PROVIDERS.rst#mixed-governance-model-for-3rd-party-related-community-providers also offer to backport features/fixes and cut releases for older versions of the providers. It's something that we never did before but we do have the capability. Thus even if we are forced to upgrade `watchtower` in case we have some important fix that we want to backport to older provider versions we can still do that.

Reply:
Let's see - as long as it does not cause back-tracking issues, it's pretty indifferent to us - and it's easier to keep it to >2, but yes, it is is IMPORTANT to keep it for AWS then we should likely make sure it is documented in more MUST WAY and likely figure out what to do in the future, because at some point in time it **might** become problematic and we need to know what to do. 

Question : how much REQUIRED watchtower is for the provider? Maybe another - better - option will be to make the watchtower depdendency optional in the provider? We have a few of those for other providers and if it is optional then we can keep it whatever ( > 3) - it will not be a requirement. It will not be installed automatically as well, but could be installed with `apache-airlfow-providers-amazon[watchtower]`  ? @shubham22 @o-nikolas ?

Reply:
Looks like no impact on backtracking after all other fixes - so we can keep it for now.